catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -577,18 +634,35 @@ protected ResourceInfo getResourceInfo(
         String metacardId = (String) value;
         LOGGER.debug("metacardId = {},   site = {}", metacardId, site);
         query = createMetacardIdQuery(metacardId);
-      }
+      } else if (ResourceRequest.GET_RESOURCE_BY_METACARD.equals(name)) {
+        // no need to actually perform the query, just create a resource info with the metacard
+        LOGGER.debug("get resource by metacard");
+        final Metacard metacard = (Metacard) value;
+        final String metacardId = metacard.getId();
 
-      QueryRequest queryRequest =
-          new QueryRequestImpl(
-              anyTag(query, site, isEnterprise),
-              isEnterprise,
-              Collections.singletonList(site == null ? this.getId() : site),
-              resourceRequest.getProperties());
+        LOGGER.debug("metacardId = {},   site = {}", metacardId, site);
+        if (!requestProperties.containsKey(Metacard.ID)) {
```

Comment:
❗ I do not think the ID case is secure.  Redacted metacards will always contain their ID.  The `RESOURCE_URI` case should be safe since the resource URI would be redacted or filtered if you did not have access to it.

If request by ID is made, we should at least check if the resource URI is present.

Reply:
For local content resources, the URI is always content:id, so in that case, it wouldn't be secure either.

Reply:
How is that different than the above branch (i.e. by metacard ID)?

Reply:
@stustison Don't we redact all fields including resource URI though even in the local content resources case?

Reply:
@paouelle The difference is before we always did a search which would filter or redact that value if the current user does not have proper access.  This new path allows a way to bypass that check so we need to make sure that the metacard they have has not been redacted of everything other than its ID and security attributes.

Reply:
but it does mean that you have gotten your hands on such a metacard through a query of your own which would have redacted the data the same way. It just moves the query request before the resource request.

Reply:
We redact the local content resource uri, yes, but as long as you have the metacard id and know that the scheme is simply "content" then you can request the resource.

Reply:
I missed the `if (resourceInfo.getResourceUri() == null) { throw new ResourceNotFoundException` just beyond the diff portion being displayed. So this would work as is as long as an endpoint never allowed a way to request a resource and provided the client a way to pass it a metacard.

Reply:
Removed this part of the PR and kept it around to re-introduce if need be later